pylint/checkers/base/basic_checker.py

```diff
@@ -453,7 +453,14 @@ def visit_expr(self, node: nodes.Expr) -> None:
 
         # Warn W0133 for exceptions that are used as statements
         if isinstance(expr, nodes.Call):
-            inferred = utils.safe_infer(expr)
+            name = ""
+            if isinstance(expr.func, nodes.Name):
+                name = expr.func.name
+            elif isinstance(expr.func, nodes.Attribute):
+                name = expr.func.attrname
+
+            # Heuristic: only run inference for names that begin with an uppercase char
+            inferred = utils.safe_infer(expr) if name[:1].isupper() else None
```

Comment:
Could you add a small note regarding the performance issue here.
Just so we know later why we added it.